docs/content/docs/ops/state/state_backends.md

```diff
@@ -325,6 +325,129 @@ public class MyOptionsFactory implements ConfigurableRocksDBOptionsFactory {
 
 {{< top >}}
 
+## Enabling Changelog
+
+// todo: Chinese version of all changed docs
+
+// todo: mention in [large state tuning]({{< ref "docs/ops/state/large_state_tuning" >}})? or 1.16?
+
+{{< hint warning >}} This feature is in experimental status. {{< /hint >}}
+
+{{< hint warning >}} Enabling Changelog may have a negative performance impact on your application (see below). {{< /hint >}}
+
+### Introduction
+
+Changelog is a feature that aims to decrease checkpointing time and, therefore, end-to-end latency in exactly-once mode.
+
+Most commonly, checkpoint duration is affected by:
+
+1. Barrier travel time and alignment, addressed by
+   [Unaligned checkpoints]({{< ref "docs/ops/state/checkpointing_under_backpressure#unaligned-checkpoints" >}})
+   and [Buffer debloating]({{< ref "docs/ops/state/checkpointing_under_backpressure#buffer-debloating" >}})
+2. Snapshot creation time (so-called synchronous phase), addressed by asynchronous snapshots
```

Comment:
`asynchronous snapshots`

What are asynchronous snapshots? and how it is different from `asynchronous phase`

Do you mean the materialization is separated and independent with the checkpointing/snapshotting?

Reply:
> What are asynchronous snapshots? and how it is different from asynchronous phase

synchronous phase = create a snapshot
asynchronous phase = upload a snapshot

Implemented naively, sync phase would duplicate **all** the state (copying all in-memory tables and/or on-disk files). So that async phase and processing can proceed concurrently.

With Asynchronous snapshots (probably a misnormer), sync phase only flushes or modifies small portions of data, as I explained [above](https://github.com/apache/flink/pull/18431#discussion_r790783796).

> Do you mean the materialization is separated and independent with the checkpointing/snapshotting?

No. As I said [above](https://github.com/apache/flink/pull/18431#discussion_r791455950), asynchronous snapshots are not related to Changelog.

Reply:
**Without Changelog:**
Sync phase: flush in-mem data -> disk, prepare what-ssts to upload

**With Changelog:**
Sync phase: prepare state-changes to upload

From the differences above, I can not infer what async snapshots refer to and can not see what problem the "async snapshots are addressed"? Or if I misunderstand the above process?

Reply:
OK, I finally got what you said.

Is there any reference you can use here? similar to Unaligned checkpoints and Buffer debloating?

It would be too confusing otherwise.

Reply:
Sure, I'll linkt to [RocksDB](https://nightlies.apache.org/flink/flink-docs-master/docs/ops/state/state_backends/#the-embeddedrocksdbstatebackend) section where they are mentioned.